mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -668,6 +668,48 @@ def SYCLCallOp : SYCL_Op<"call", [CallOpInterface]> {
   }];
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// accessor.get_pointer OPERATION
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLAccessorGetPointerOp
+    : SYCLMethodOpInterfaceImpl<"accessor.get_pointer", "AccessorType",
+                                ["get_pointer"]> {
+  let summary = "Call to accessor::get_pointer";
+  let description = [{
+    This operation represents a call to the accessor::get_pointer function.
+  }];
+
+  let arguments = (ins Arg<AccessorMemRef, "The accessor", [MemRead]>:$Acc,
+                       TypeArrayAttr:$ArgumentTypes,
+                       FlatSymbolRefAttr:$FunctionName,
+                       OptionalAttr<FlatSymbolRefAttr>:$MangledFunctionName,
+                       FlatSymbolRefAttr:$TypeName);
+
+  let results = (outs SYCL_MultiPtrType:$result);
```

Comment:
When I compile `get_pointer()`, we currently generate a call that returns `!sycl_multi_ptr_i32_1_`
```
%19 = sycl.call @get_pointer(%18) {MangledFunctionName = @_ZNK4sycl3_V18accessorIfLi2ELNS0_6access4modeE1025ELNS2_6targetE2014ELNS2_11placeholderE0ENS0_3ext6oneapi22accessor_property_listIJEEEE11get_pointerILS4_2014EvEENS0_9multi_ptrIfLNS2_13address_spaceE1ELNS2_9decoratedE2EEEv, TypeName = @accessor} : (memref<?x!sycl_accessor_2_f32_w_gb, 4>) -> !sycl_multi_ptr_f32_1_
```

Reply:
I see. DPC++ has these deprecated functions defined in the headers and they're shadowing the 2020 version returning a pointer. I'd rather have the non-deprecated one in the dialect, as this functionality is now available through `get_multi_ptr`. 

WDYT?

Reply:
I agree, the multi-ptr is now available through an explicit API and we should model SYCL 2020 rather than deprecated features. 

Would the output change if there was a language version macro defined?

Reply:
Why? This is not something you would or even could use in host code.

Reply:
Currently when I build the SYCL RT (I preprocessed the accessor.hpp file using the command line invocation the cmake file generates) I see: 
```
  DataT *get_pointer() const

  {
    return getPointerAdjusted();
  }
``` 
So the DPC++ compiler builds the SYCL RT with the "old" SYCL APIs.

Reply:
For device, DPC++ always generate
```
  template <
      access::target AccessTarget_ = AccessTarget,
      typename = detail::enable_if_t<AccessTarget_ == access::target::device>>
  global_ptr<DataT> get_pointer() const {
    return global_ptr<DataT>(getPointerAdjusted());
  }
```

Reply:
yes.